[Place Page SEO] Add meta tag to prevent indexing on non-prod place pages

juliawu opened the pull request:
This PR adds `<meta name="robots" content="noindex">` as to place pages when not on the main `https://datacommons.org` domain to [prevent search crawlers from indexing](https://developers.google.com/search/docs/crawling-indexing/block-indexing).

This means only the `https://datacommons.org/place/` pages will get indexed, and the crawler will ignore Custom DCs and alternate domains like `browser.datacommons.org` and `dev.datacommons.org`. This will reduce the spurious indexing error alerts for pages that shouldn't be indexed.

dwnoble commented:
@juliawu can you verify this works as expected in a test GCP environment? I suspect the base_url might not get set properly when going from the gcp load balancer -> kubernetes cluster

juliawu commented:
> @juliawu can you verify this works as expected in a test GCP environment? I suspect the base_url might not get set properly when going from the gcp load balancer -> kubernetes cluster

Just tested by deploying to dev.datacommons.org. `base_url` does get set to `http://datacommons.org` (note HTTP instead of HTTPS). Switched the logic to allow both HTTP and HTTPS for now since I'm not sure which base_url will give when deployed to the prod server.

dwnoble commented:
> > @juliawu can you verify this works as expected in a test GCP environment? I suspect the base_url might not get set properly when going from the gcp load balancer -> kubernetes cluster
> 
> Just tested by deploying to dev.datacommons.org. `base_url` does get set to `http://dev.datacommons.org` (note HTTP instead of HTTPS). Switched the logic to allow both HTTP and HTTPS for now since I'm not sure which base_url will give when deployed to the prod server.

Great! So we can understand what the URL looks like in prod, can you add some sort of logging (maybe another meta tag directly under the noindex tag? `<meta name="dc-debug" content="{url}">`. Also-can you update update the logic to not show the noindex tag in staging (so it properly mimics the prod env)?

dwnoble commented on static/package-lock.json:
Nit: remove package-lock.json changes

juliawu replied:
Removed!

juliawu commented:
> > > @juliawu can you verify this works as expected in a test GCP environment? I suspect the base_url might not get set properly when going from the gcp load balancer -> kubernetes cluster
> > 
> > 
> > Just tested by deploying to dev.datacommons.org. `base_url` does get set to `http://dev.datacommons.org` (note HTTP instead of HTTPS). Switched the logic to allow both HTTP and HTTPS for now since I'm not sure which base_url will give when deployed to the prod server.
> 
> Great! So we can understand what the URL looks like in prod, can you add some sort of logging (maybe another meta tag directly under the noindex tag? `<meta name="dc-debug" content="{url}">`. Also-can you update update the logic to not show the noindex tag in staging (so it properly mimics the prod env)?

Good call outs. Added logging statements we can view in GKE logging (instead of adding meta tag end users also see).  Also updated the logic to not show noindex tag in staging.

dwnoble reviewed (approved):
Thank you!
